### PR TITLE
Adding nosniff Header

### PR DESCRIPTION
**Pull request by linkRace**

### CODE REVIEW

File: lib/nosniff.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * https://blogs.msdn.microsoft.com/ie/2008/09/02/ie8-security-part-vi-beta-2-update/
+ */
+module.exports = function nosniff(options) {
+    options = options || {};
+
+    // `enabled` should be either `1` or `0`
+    var enabled = (options.enabled !== undefined) ? +options.enabled : 1;
+    
+    return function xssProtection(req, res, next) {
+        if (enabled) {
```
Comment (jasisk) [inline lib/nosniff.js]: the `options` are fixed—there is no need for this execution-time conditional.

Comment (linkRace) [inline lib/nosniff.js]: Was trying to conform it to the rest of the code base, wasn't quite sure how it all worked out.  Does the enabled part not refer to setting it true/false in config?

Comment (jasisk) [inline lib/nosniff.js]: It does but `options.enabled` will never change. That conditional could be pushed up and out of the middleware itself. Of course, that would mean:

``` js
if (enabled) {
  return function xssProtection(req, res, next) {
    // set header
    next();
  }
} else {
  return function noop(req, res, next) {
    next();
  }
}
```

Pretty pedantic but `¯\_(ツ)_/¯`

Comment (jasisk) [inline lib/nosniff.js]: Ack. In fact, there isn't a need for any options at all.

Comment (linkRace) [inline lib/nosniff.js]: So just
module.exports = function nosniff(enabled) {
    if (enabled) {
      return function xssProtection(req, res, next) {
        res.header('X-Content-Type-Options', 'nosniff');
        next();
      };
    } else {
       return function noop(req, res, next) {
          next();
       }
    }
};

**Comment by jasisk** [comment]
``` js
module.exports = function nosniff() {
    return function xssProtection(req, res, next) {
        res.header('X-Content-Type-Options', 'no sniff');
        next();
    };
}

**Comment by linkRace** [comment]
xssProtection?

**Comment by jasisk** [comment]
> xssProtection?

edited. C/P from your code.

**Comment by linkRace** [comment]
Cool, updated file.

**Comment by jasisk** [comment]
Throw some documentation in the README and I'm a :+1:.

**Comment by linkRace** [comment]
Cool, combine this with the csp update to make a 1.3.1 release I'd think.

**Comment by grawk** [comment]
Looks like the CI job is failing on the jshint check. Also, it'd be a good idea to update the .travis.yml config  to just test on node 4 and maybe node 0.12

**Comment by grawk** [comment]
👍

File: README.md [hunk 2]
```diff
@@ -43,6 +44,7 @@ app.use(lusca.xframe('SAMEORIGIN'));
 app.use(lusca.p3p('ABCDEF'));
 app.use(lusca.hsts({ maxAge: 31536000 }));
 app.use(lusca.xssProtection(true));
+app.use(lusca.nosniff(true));
```
Comment (jasisk) [inline README.md]: This method doesn't take any arguments.

**Comment by jasisk** [comment]
This needs tests, no?

**Comment by tlivings** [comment]
Got merged without 2 :+1: ?

**Comment by indutny** [comment]
👍

**Pull request merged**